test/Transactions/Orleans.Transactions.Tests/Grains/MultiStateTransactionalBitArrayGrain.cs

```diff
@@ -0,0 +1,114 @@
+﻿
+using System;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using Orleans.Transactions.Abstractions;
+
+namespace Orleans.Transactions.Tests.Correctness
+{
+    [Serializable]
+    public class BitArrayState
+    {
+        private static readonly int BitsInInt = sizeof(int) * 8;
+
+        private int[] value = new int[] { 0 };
+        public int[] Value => value;
+
+        public void Set(int index, bool value)
+        {
+            if (index >= this.value.Length * BitsInInt)
+            {
+                Array.Resize(ref this.value, this.value.Length + 1);
+            }
+            int idx = this.value.Length / BitsInInt;
+            int shift = 1 >> (this.value.Length % BitsInInt);
+            if (value)
+            {
+                this.value[idx] |= shift;
+            } else
+                this.value[idx] &= ~shift;
+        }
+    }
```

Comment:
the shift is definitely wrong.  The rest is ok I think.  Take another look.

Reply:
If the rest is right, I must misunderstand what this does. My interpretation was that this is essentially a bit vector which internally grows in size automatically depending on how far into it we access.

Eg, I can create a new instance if it and then set bit 1024 if I like. The current implementation only allows growing by 32 bits at a time.

The shift direction and magnitude is wrong, and so is the index... Unless I'm misunderstanding

Reply:
Nope, you're correct.  This is a mess :/

Reply:
Since this is not perf-critical code, why not just use a Dictionary<int,bool>? Were you hoping to reuse the bitvector in other places? But then it probably needs unit tests.

Reply:
I'd intended to reuse some of this in the correctness load testing, once we get past functional, so the bit field needs be compact enough to store thousands of bits, without getting tot bloated in storage.